File: tests/fixtures/edgecase_repo/swift/Service2.swift
import Foundation

protocol Handler2 {
    func handle(_ payload: String) -> Bool
}

class Base2 {
    let name: String
    init(name: String) {
        self.name = name
    }
}

class Service2: Base2, Handler2 {
    func handle(_ payload: String) -> Bool {
        self.validate(payload)
        return true
    }

    private func validate(_ payload: String) {}
}

func helper2(_ value: String) -> String {
    return value
}

File: tests/fixtures/edgecase_repo/scala/Service2.scala
package com.example.gen

import com.example.gen.Base2

trait Handler2 {
  def handle(payload: String): Boolean
}

class Base2(val name: String)

class Service2(name: String) extends Base2(name) with Handler2 {
  def handle(payload: String): Boolean = {
    this.validate(payload)
    true
  }

  private def validate(payload: String): Unit = {}
}

object Helpers2 {
  def helper(value: String): String = value
}
